docs/DeveloperGuide.md

```diff
@@ -221,47 +216,61 @@ When the `EventCreateCommand` finishes executing, the updated `EventStorage` is
     * Cons:
       * A new method must be created to save data after executing other commands. E.g. The save() method after an edit event command would be different from a create event command as the `Event` would have to be located first in the JSON file and then updated.
 
-### \[Proposed\] Undo/redo feature
+### Undo/redo feature
 
 #### Proposed Implementation
 
-The proposed undo/redo mechanism is facilitated by `VersionedAddressBook`. It extends `AddressBook` with an undo/redo history, stored internally as an `addressBookStateList` and `currentStatePointer`. Additionally, it implements the following operations:
+The proposed undo/redo mechanism is facilitated by `VersionedVolunteerStorage` and `VersionedEventStorage`. It extends `VolunteerStorage` and `EventStorage` respectively with an undo/redo history, stored internally as a `versionedVolunteers`, `versionedEvents` and `currentStatePointer`. 
+
+`VersionedVolunteerStorage` implements the following operations:
+
+* `VersionedVolunteerStorage#saveNewState()` — Saves the current volunteers state in its history.
+* `VersionedVolunteerStorage#undo()` — Restores the previous volunteers state from its history.
+* `VersionedVolunteerStorage#redo()` — Restores a previously undone volunteers state from its history.
+
+`VersionedEventStorage` implements the following operations:
 
-* `VersionedAddressBook#commit()` — Saves the current address book state in its history.
-* `VersionedAddressBook#undo()` — Restores the previous address book state from its history.
-* `VersionedAddressBook#redo()` — Restores a previously undone address book state from its history.
+* `VersionedEventStorage#saveNewState()` — Saves the current events state in its history.
+* `VersionedEventStorage#undo()` — Restores the previous events state from its history.
+* `VersionedEventStorage#redo()` — Restores a previously undone events state from its history.
+* 
 
-These operations are exposed in the `Model` interface as `Model#commitAddressBook()`, `Model#undoAddressBook()` and `Model#redoAddressBook()` respectively.
+These operations are exposed in the `Model` interface as `Model#commitToBothVersionedStorages()`, `Model#undoBothStorages()` and `Model#redoBothStorages()` respectively.
+
+<box type="info" seamless>
+
+**Note:** Any command that changes the volunteers state or events state will trigger the `saveNewState()` method for both `VersionedVolunteerStorage` and `VersionedEventStorage`. Similarly, a `undo` or `redo` command will trigger the `undo()` and `redo()` method for both `VersionedVolunteerStorage` and `VersionedEventStorage`. To avoid diagram repetition, I will mainly refer to `VersionedVolunteerStorage` in the following illustrations. However, take note that whatever happens at `VersionedVolunteerStorage` occurs for `VersionedEventStorage` as well.
```

Comment:
I think the "I will mainly refer to..." part sounds abit personal hahaha. maybe can make it sound like third person perspective?

Reply:
oki done